project-backend.Test/ControllerTests/JobControllerTest.cs

```diff
@@ -0,0 +1,242 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging.Abstractions;
+using Moq;
+using project_backend.Controllers;
+using project_backend.Models.Job;
+using project_backend.Models.JobController.GetJobs;
+using project_backend.Models.User;
+using project_backend.Providers.JobProvider;
+using project_backend.Repos;
+using System;
+using System.Reflection;
+using System.Security.Claims;
+using Xunit;
+using Xunit.Abstractions;
+using Xunit.Sdk;
+
+namespace project_backend.Test.ControllerTests
+{
+    [Trait("Category", "Unit")]
+    public class JobControllerTest : IDisposable
+    {
+        private readonly JobController _controller;
+        private readonly DatabaseContext _dbContext;
+
+        public JobControllerTest(ITestOutputHelper output)
+        {
+            var type = output.GetType();
+            var testMember = type.GetField("test", BindingFlags.Instance | BindingFlags.NonPublic);
+            var test = (ITest)testMember.GetValue(output);
+
+            var options = new DbContextOptionsBuilder<DatabaseContext>()
+                .UseInMemoryDatabase(test.DisplayName) // use the name of the test, because otherwise the db is used simultaneously by multiple threads
+                .Options;
+            _dbContext = new DatabaseContext(options);
+
+            var user1 = new UserDAO
+            {
+                Email = "user1@gmail.com",
+                DateOfBirth = new DateTime(2020, 2, 20),
+                FirstName = "User1",
+                LastName = "User1",
+                Password = ""
+            };
+
+            var user2 = new UserDAO
+            {
+                Email = "user2@gmail.com",
+                DateOfBirth = new DateTime(2021, 1, 1),
+                FirstName = "User2",
+                LastName = "User2",
+                Password = ""
+            };
+
+            _dbContext.Users.AddRange(user1, user2);
+            _dbContext.SaveChanges();
+
+            _dbContext.Jobs.AddRange(
+                    new JobDAO
+                    {
+                        Name = "Job1",
+                        Description = "Job1Desc",
```

Comment:
For tests in general you can use the arrange/act/assert setup. It's usual for backend testing to be done this way. It's not necessary but it just formalizez how tests look. Also, you are testing the controller here, but you also use the provider, so you're basically testing that as well, making it more of an integration test than an unit test. You should instead just mock the provider to return what it has to, without having to create a database context and the sort.

Reply:
Fixed in https://github.com/PC-Chair-Fight/project-backend/pull/17/commits/8e24bcc8257ebf4c10069b01a9af7a3efb14f3ee